projects/vdk-core/src/vdk/internal/builtin_plugins/run/summary_output.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import json
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import HookCallResult
+from vdk.internal.builtin_plugins.run.execution_results import ExecutionResult
+from vdk.internal.builtin_plugins.run.execution_results import StepResult
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core.config import ConfigurationBuilder
+
+log = logging.getLogger(__name__)
+
+JOB_RUN_SUMMARY_FILE_PATH = "JOB_RUN_SUMMARY_FILE_PATH"
+
+
+class JobRunSummaryOutputPlugin:
+    @hookimpl
+    def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
+        config_builder.add(
+            key=JOB_RUN_SUMMARY_FILE_PATH,
+            default_value="",
+            description="The path location of the file where job run result summary is stored.",
+        )
+
+    @hookimpl(hookwrapper=True)
+    def run_job(self, context: JobContext) -> None:
+        out: HookCallResult
+        out = yield
```

Comment:
Nit: This pulls the execution result out, right? Might be worth adding a comment.